### DISCUSSION TITLE
OpenAPI generation and Deduced Environments

### DISCUSSION THREAD

**Discussion by scprek** [micronaut-projects/micronaut-openapi#1493] [General]
Hi,

When I do a gradle build locally all of my controller routes show up in the generated OpenApi yaml docs. Even though most controllers have a `@Requires` annotation, I don't need to specify which environments if I want the docs to have all of the routes. I only specify the environments if I want to limit, say create internal/external docs.

However, in GitHub Actions, the opposite is true. The OpenApi docs only contain the title/top level OpenApi info but no routes/schemas. I was able to fix this by explicitly stopping deduction of environments in the build step. Is it expected that the deduced environment `CLOUD` honors the `@Requires` annotations and expects explicit environment setting when locally it does not?

### Cloud Env Empty Docs

`MICRONAUT_ENVIRONMENTS=CLOUD ./gradlew :kaptKotlin`

```
❯ cat build/tmp/kapt3/classes/main/META-INF/swagger/*.yml
openapi: 3.0.1
info:
  title: test-app
  description: Reference app
  version: 0.2.0
```

### Fix in CI

`./gradlew -Dmicronaut.env.deduction=false :kaptKotlin`

**Comment by altro3** [comment]
If you have suggestion to improvements - create issues, please

**Reply by scprek** [reply]
I guess I was more curious if that’s just how it should work or not first and then can open up an issue to discuss more. If disabling deduction for my use case of just wanting to generate the docs in a CI build to publish somehow is correct then no changes needed.
